src/fabric/src/fabric_doc_update.erl

```diff
@@ -39,10 +39,15 @@ go(DbName, AllDocs0, Opts) ->
     Acc0 = {length(Workers), length(AllDocs), list_to_integer(W), GroupedDocs, dict:new()},
     Timeout = fabric_util:request_timeout(),
     try rexi_utils:recv(Workers, #shard.ref, fun handle_message/3, Acc0, infinity, Timeout) of
+        {ok, {_, []}} ->
+            erlang:error({internal_server_error, "No Responses For Document Update"});
         {ok, {Health, Results}} when
             Health =:= ok; Health =:= accepted; Health =:= error
         ->
-            {Health, [R || R <- couch_util:reorder_results(AllDocs, Results), R =/= noreply]};
+            {Health, [
+                R
+             || R <- couch_util:reorder_results(AllDocs, Results, timeout), R =/= noreply
+            ]};
```

Comment:
I had noticed below we already defined an error for missing results 
https://github.com/apache/couchdb/blob/02fb6e513149377bb35d1559f7d2cf38c06e2dad/src/fabric/src/fabric_doc_update.erl#L147-L148

There is test for it below too, I believe.

We can probably re-use the same internal_error response.

Another improvement we could make is make sure Health becomes an `error` if at least one document is missing. This could be a bit tricky here depending how we do it. We could use a custom version of reorder_results for this module only (reorder_final_results?) and make sure to also update the health to error if we encounter at least one `internal_server_error`. Another approach could be to run through `couch_util:reorder_results(AllDocs, Results, {error, internal_server_error})` and then, as a separate step make sure to check all response and make sure to turn Health into an `error`

Reply:
I went with the latter solution because there is a `fabric` update test that mecks `couch_util:reorder_results` and for some reason it wasn't`meck:expect` didn't meck `fabric_update_doc:reorder_results` correctly when I put  it in that module. We'd have to also expose the method for mecking. this way, we don't need to export the function in `fabric_update_doc` and instead we keep it compartmentalized in `couch_util` for exporting.